File: verify_node_finance.py
import requests
import json

API_BASE = 'http://localhost:5000/api/v1'

def verify():
    print('🔍 Verifying Node Financial Endpoints (Python)...')
    
    try:
        # Login
        login_payload = {
            'email': 'admin@storeai.com',
            'password': 'AdminPassword123!'
        }
        res = requests.post(f"{API_BASE}/auth/login", json=login_payload)
        res.raise_for_status()
        
        token = res.json().get('token')
        headers = {'Authorization': f"Bearer {token}"}
        
        print('✅ Login Successful')

        # 1. Balance Sheet
        bs_res = requests.get(f"{API_BASE}/finance/balance-sheet", headers=headers)
        bs_res.raise_for_status()
        print('📊 Balance Sheet Content Received')
        print(json.dumps(bs_res.json(), indent=2))

        # 2. Ledger fallback check
        ledger_res = requests.get(f"{API_BASE}/accounts/ledger", headers=headers)
        ledger_res.raise_for_status()
        entries = ledger_res.json()
        print(f"📜 Ledger Entries Found: {len(entries)}")
        
        if len(entries) > 0:
            print(f"   Sample Entry: {entries[0].get('title')}")

        print('✨ Verification Complete!')
    except Exception as e:
        print(f"❌ Verification Failed: {str(e)}")
        if hasattr(e, 'response') and e.response:
            print(f"   Error Detail: {e.response.text}")

if __name__ == "__main__":
    verify()

File: verify_node_finance.ts
import axios from 'axios';

const API_BASE = 'http://localhost:5000/api/v1';

async function verify() {
    console.log('🔍 Verifying Node Financial Endpoints...');

    try {
        // We need a token. I'll try to login with the admin user from .env
        const loginRes = await axios.post(`${API_BASE}/auth/login`, {
            email: 'admin@storeai.com',
            password: 'AdminPassword123!'
        });

        const token = loginRes.data.token;
        const config = { headers: { Authorization: `Bearer ${token}` } };

        console.log('✅ Login Successful');

        // 1. Balance Sheet
        const bsRes = await axios.get(`${API_BASE}/finance/balance-sheet`, config);
        console.log('📊 Balance Sheet:', JSON.stringify(bsRes.data, null, 2));

        // 2. Ledger fallback check
        const ledgerRes = await axios.get(`${API_BASE}/accounts/ledger`, config);
        console.log(`📜 Ledger Entries Found: ${ledgerRes.data.length}`);
        if (ledgerRes.data.length > 0) {
            console.log('   Sample Entry:', ledgerRes.data[0].title);
        }

        console.log('✨ Verification Complete!');
    } catch (e: any) {
        console.error('❌ Verification Failed:', e.response?.data || e.message);
    }
}

verify();
